rdt-open-reader/src/test/java/io/ona/rdt/robolectric/util/UtilsTest.java

```diff
@@ -223,4 +228,16 @@ public void testIsValidJSONObjectShouldReturnCorrectStatus() {
         Assert.assertFalse(Utils.isValidJSONObject(null));
         Assert.assertFalse(Utils.isValidJSONObject(""));
     }
+
+    @Test
+    public void testVerifyUserAuthorizationShouldCallOnlyOnce() throws AuthenticatorException, OperationCanceledException, IOException {
+
+        SyncUtils syncUtils = Mockito.mock(SyncUtils.class);
+        Mockito.when(syncUtils.verifyAuthorization()).thenReturn(false);
+        Utils.verifyUserAuthorization(syncUtils);
```

Comment:
Also test what happens when the asynctask returns a state of `FINISHED`.

Reply:
done